bitcoin/script/miniscript.h

```diff
@@ -1637,10 +1639,8 @@ inline NodeRef<Key> DecodeScript(I& in, I last, const Ctx& ctx)
 template<typename Ctx>
 inline NodeRef<typename Ctx::Key> FromString(const std::string& str, const Ctx& ctx) {
     using namespace internal;
-    Span<const char> span = MakeSpan(str);
-    auto ret = Parse<typename Ctx::Key>(span, ctx);
-    if (!ret) return {};
-    return ret;
+    Span<const char> span = Span{str};
+    return Parse<typename Ctx::Key>(span, ctx);
```

Comment:
I think this can be written even more concisely as `return internal::Parse<typename Ctx::Key>(str, ctx);`.

Reply:
... Which makes the introduction of the new `span.h` needless :)

I'll make the change in #79 and close this PR then.

Reply:
Ah, right.

Reply:
Oh, actually not! It is enabled by the `span.h` update

Reply:
So in any case i get a compilation error:
```
g++ -O3 -g0 -Wall -std=c++17 -march=native -flto -Ibitcoin bitcoin/util/strencodings.cpp bitcoin/util/spanparsing.cpp bitcoin/script/script.cpp bitcoin/script/miniscript.cpp compiler.cpp main.cpp -o miniscript
In file included from main.cpp:7:
bitcoin/script/miniscript.h: In instantiation of ‘miniscript::NodeRef<typename Ctx::Key> miniscript::FromString(const string&, const Ctx&) [with Ctx = CompilerContext; miniscript::NodeRef<typename Ctx::Key> = std::shared_ptr<const miniscript::Node<std::__cxx11::basic_string<char> > >; typename Ctx::Key = std::__cxx11::basic_string<char>; std::string = std::__cxx11::basic_string<char>]’:
main.cpp:23:72:   required from here
bitcoin/script/miniscript.h:1641:46: error: no matching function for call to ‘Parse<CompilerContext::Key>(const string&, const CompilerContext&)’
 1641 |     return internal::Parse<typename Ctx::Key>(str, ctx);
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
bitcoin/script/miniscript.h:912:21: note: candidate: ‘template<class Key, class Ctx> miniscript::NodeRef<Key> miniscript::internal::Parse(Span<const char>&, const Ctx&)’
  912 | inline NodeRef<Key> Parse(Span<const char>& in, const Ctx& ctx)
      |                     ^~~~~
bitcoin/script/miniscript.h:912:21: note:   template argument deduction/substitution failed:
bitcoin/script/miniscript.h:1641:46: note:   cannot convert ‘str’ (type ‘const string’ {aka ‘const std::__cxx11::basic_string<char>’}) to type ‘Span<const char>&’
 1641 |     return internal::Parse<typename Ctx::Key>(str, ctx);
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
make: *** [Makefile:5: miniscript] Error 1
```

Reply:
Ok, that's because `Parse` takes a mutable reference as input. The simple solution is to make it take a by-value argument instead (`Span<const char> in`).

Reply:
Thanks, done. I also checked and it doesn't compile on #79.